Release and integrate Shoal 3.1.0

I have noticed that we forgot to integrate fixes from shoal. This stacktrace was fixed in commit 4edb578688999397ae93762342005b5b0a000291 of https://github.com/eclipse-ee4j/glassfish-shoal/pull/166 
All changes: https://github.com/eclipse-ee4j/glassfish-shoal/milestone/5?closed=1

```
[2024-01-10T10:39:01.983105+01:00] [GF 7.0.12-SNAPSHOT] [SEVERE] [NCLS-CLSTR-10117] [jakarta.enterprise.cluster.gms] [tid: _ThreadID=217 _ThreadName=pool-24-thread-1] [levelValue: 1000] [[
GMS failed to start. See stack trace for additional information.
com.sun.enterprise.ee.cms.core.GMSException: failed to join group ClusterOne
at com.sun.enterprise.ee.cms.impl.base.GMSContextImpl.join(GMSContextImpl.java:171)
at com.sun.enterprise.ee.cms.impl.common.GroupManagementServiceImpl.join(GroupManagementServiceImpl.java:335)
at org.glassfish.gms.GMSAdapterImpl.initializeGMS(GMSAdapterImpl.java:877)
at org.glassfish.gms.GMSAdapterImpl.initialize(GMSAdapterImpl.java:366)
at org.glassfish.gms.bootstrap.GMSAdapterService.loadModule(GMSAdapterService.java:217)
at org.glassfish.gms.bootstrap.GMSAdapterService.checkCluster(GMSAdapterService.java:191)
at org.glassfish.gms.bootstrap.GMSAdapterService$1.changed(GMSAdapterService.java:248)
at org.jvnet.hk2.config.ConfigSupport.sortAndDispatch(ConfigSupport.java:294)
at org.glassfish.gms.bootstrap.GMSAdapterService.changed(GMSAdapterService.java:240)
at org.jvnet.hk2.config.Transactions$ConfigListenerJob.process(Transactions.java:407)
at org.jvnet.hk2.config.Transactions$ConfigListenerJob.process(Transactions.java:396)
at org.jvnet.hk2.config.Transactions$ConfigListenerNotifier$1$1.call(Transactions.java:285)
at org.jvnet.hk2.config.Transactions$ConfigListenerNotifier$1$1.call(Transactions.java:282)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NoClassDefFoundError: org/glassfish/grizzly/attributes/NullaryFunction
at com.sun.enterprise.mgmt.transport.grizzly.grizzly2.GrizzlyNetworkManager2.initialize(GrizzlyNetworkManager2.java:178)
at com.sun.enterprise.mgmt.ClusterManager.<init>(ClusterManager.java:109)
at com.sun.enterprise.ee.cms.impl.base.GroupCommunicationProviderImpl.initializeGroupCommunicationProvider(GroupCommunicationProviderImpl.java:128)
at com.sun.enterprise.ee.cms.impl.base.GMSContextImpl.join(GMSContextImpl.java:166)
... 16 more
Caused by: java.lang.ClassNotFoundException: org.glassfish.grizzly.attributes.NullaryFunction not found by org.glassfish.main.grizzly.nucleus-grizzly-all [53]
at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1591)
at org.apache.felix.framework.BundleWiringImpl.access$300(BundleWiringImpl.java:79)
at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:1976)
at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
at org.apache.felix.framework.BundleWiringImpl.getClassByDelegation(BundleWiringImpl.java:1358)
at org.apache.felix.framework.BundleWiringImpl.searchImports(BundleWiringImpl.java:1612)
at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1528)
at org.apache.felix.framework.BundleWiringImpl.access$300(BundleWiringImpl.java:79)
at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:1976)
at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
... 20 more
]]
```

--- a/nucleus/parent/pom.xml
+++ b/nucleus/parent/pom.xml
@@ -127,7 +127,7 @@
         <glassfish-management-api.version>3.2.3</glassfish-management-api.version>
         <pfl.version>4.1.2</pfl.version>
         <gmbal.version>4.0.3</gmbal.version>
-        <shoal.version>3.0.1</shoal.version>
+        <shoal.version>3.1.0</shoal.version>
         <ha-api.version>3.1.13</ha-api.version>
         <logging-annotation-processor.version>1.9</logging-annotation-processor.version>
         <command.security.maven.plugin.isFailureFatal>true</command.security.maven.plugin.isFailureFatal>

